src/placement/ring_map.c

```diff
@@ -1253,8 +1253,10 @@ ring_obj_find_reint(struct pl_map *map, struct daos_obj_md *md,
 			return rc;
 		rc = pl_obj_layout_alloc(rop.rop_grp_size, rop.rop_grp_nr,
 				&reint_layout);
-		if (rc)
+		if (rc) {
+			D_FREE(layout);
```

Comment:
I just reviewed this same patch against 2.0.  See the comments there.   The process for this is to get patch landed on master before cherry picking it to 2.0.  I don't see much reason to diverge from this here.

Anyway, the comment I made here on the other is that this should be free'd inside of pl_obj_layout_alloc if it returns an error.  If it isn't, it's a bug in that routine.

Reply:
The issue is layout is successful but the second pl_obj_layout_alloc fails leaving layout leaked.

Reply:
ah, I see. I missed that it was for layout.